languages.adoc

```diff
@@ -18,88 +18,102 @@ SPDX-FileType: DOCUMENTATION
 SPDX-License-Identifier: Apache-2.0
 ----
 
-== Overview
+== Overview 
 
-Language specific uProtocol libraries (_the library_) implement the data model defined in link:../basics/README.adoc[uProtocol Basic Types] in a particular programming language like Java, Rust or C/C++. A library also provides helpers for validating data model instances, serializing UUris to a textual representation as well as builders for correctly instantiating more complex data types. Each library includes the language specific declaration of the abstract interfaces defined in uProtocol link:../up-l1/README.adoc[Layer 1] and link:../up-l2/README.adoc[Layer 2].
+uProtocol Language specific libraries (_libraries_) are written for a given programming language (ex. Java, C++, etc...) to implement the protocol and are the primary library that is used by applications and services (uEs) to communicate with other uEs. The _libraries_ declare the API for the uProtocol Transport Layer (uP-L1) and implement the uProtocol Communication Layer API (uP-L2).
 
-NOTE: All libraries *MUST* be made available under a permissive (OSS) license like Apache 2.0 or MIT and *MUST* be hosted under the https://github.com/eclipse-uprotocol[Eclipse uProtocol GitHub Organization].
 
-== Source Code Structure
+== Requirements
 
-The following sections define the content and structure of these libraries.
+=== Content
 
-* The root namespace (package name) *MUST* be `org.eclipse.uprotocol`
-* Modules defined in table <<content>> below *MUST* be implemented in their own folder (namespace)
-* Each module *SHOULD* separate the implementation of the specification as defined in table <<folders>> below.
+[oft-sid="req~up-language-api~1"]
+--
+* *MUST* implement serializers, validators, and builders for the link:../basics/README.adoc[uProtocol data types] using the link:../up-core-api/README.adoc[up-core-api data model].
+--
 
+[oft-sid="req~up-language-transport-api~1"]
+--
+* *MUST* declare the language specific link:up-l1/README.adoc[uProtocol Transport Layer interface], such that the interface can be implemented by the transport libraries.
+--
 
-.Module Folders/Namespace
-[#folders,width="80%",cols="20%,80%",options="header"]
-|===
-| Folder | Description
+[oft-sid="req~up-language-comm-api~1"]
+--
+* *MUST* provide an implementation of the link:up-l2/README.adoc[uProtocol Communication Layer API] based on the uProtocol Transport Layer API.
+--
```

Comment:
should it not also include an implementation of the L2 interfaces (that relies on transport API)